voltdb/datadog_checks/voltdb/check.py

```diff
@@ -0,0 +1,148 @@
+            self.register_secret(password)
+
+        manager_queries = [
+            queries.CPUMetrics,
+            queries.MemoryMetrics,
+            queries.SnapshotStatusMetrics,
+            queries.CommandLogMetrics,
+            queries.ProcedureMetrics,
+            queries.LatencyMetrics,
+            queries.StatementMetrics,
+            queries.GCMetrics,
+            queries.IOStatsMetrics,
+            queries.TableMetrics,
+            queries.IndexMetrics,
+        ]
+
+        if BASE_PARSED_VERSION < pkg_resources.parse_version('15.0.0'):
+            # On Agent < 7.24.0 we must to pass `Query` objects instead of dicts.
+            manager_queries = [Query(query) for query in manager_queries]  # type: ignore
+
+        self._query_manager = QueryManager(
+            self,
+            self._execute_query_raw,
+            queries=manager_queries,
+            tags=self._config.tags,
+        )
+        self.check_initializations.append(self._query_manager.compile_queries)
+
+    def _get_system_information(self):  # Isolated for unit testing purposes.
+        # type: () -> requests.Response
+        # See: https://docs.voltdb.com/UsingVoltDB/sysprocsysteminfo.php#sysprocsysinforetvalovervw
+        url = self._config.api_url
+        auth = self._config.auth
+        params = self._config.build_api_params(procedure='@SystemInformation', parameters=['OVERVIEW'])
+
+        try:
+            return self.http.get(url, auth=auth, params=params)
+        except Exception:
+            raise
+
+    def _fetch_version(self):
+        # type: () -> Optional[str]
+        try:
+            r = self._get_system_information()
+        except Exception:
+            raise
+
+        try:
+            r.raise_for_status()
+        except Exception as exc:
+            message = 'Error response from VoltDB: {}'.format(exc)
+            try:
+                # Try including detailed error message from response.
+                details = r.json()['statusstring']
+            except Exception:
+                pass
+            else:
+                message += ' (details: {})'.format(details)
+            raise_from(Exception(message), exc)
+
+        data = r.json()
+        rows = data['results'][0]['data']  # type: List[tuple]
+
+        # NOTE: there will be one VERSION row per server in the cluster.
+        # Arbitrarily use the first one we see.
+        for _, column, value in rows:
+            if column == 'VERSION':
+                return self._transform_version(value)
+
+        self.log.debug('VERSION column not found: %s', [column for _, column, _ in rows])
+        return None
+
+    def _transform_version(self, raw):
+        # type: (str) -> str
+        # VoltDB does not include .0 patch numbers (eg 10.0, not 10.0.0).
+        # Need to ensure they're present so the version is always in 3 parts: major.minor.patch.
+        major, rest = raw.split('.', 1)
+        minor, found, patch = rest.partition('.')
```

Comment:
Should we try to catch a `ValueError` in case the raw version is invalid since the check shouldn't fail on trying to transform version?

Reply:
Yes, good point. The only possible issue would be with the `raw.split()` call — `.partition()` can't fail.